### PR TITLE
Improve runtime of aws-iam-password-reset

### PR DESCRIPTION
**Pull request by janboll**
Remove saas files from roles query
Initialize AWS client only if it is required

https://issues.redhat.com/browse/APPSRE-7583

### CODE REVIEW

File: reconcile/aws_iam_password_reset.py [hunk 1]
```diff
@@ -47,27 +47,35 @@ def run(dry_run, defer=None):
...
+
+            if aws_api is None:
+                aws_api = AWSApi(1, [a], settings=settings)
+
+            aws_api.reset_password(account_name, user_name)
+            aws_api.reset_mfa(account_name, user_name)
+            state.add(state_key)
+
+        if aws_api:
+            aws_api.cleanup()
```
Comment (geoberle) [inline reconcile/aws_iam_password_reset.py:81]: this cleanup might not happen if the integration fails for whatever reason before it reaches this step

Comment (janboll) [reply reconcile/aws_iam_password_reset.py:81]: Right, this should be fine, the cleanup is removing sessions. 
Nevertheless, I'll change it slightly to use context-managers again

**Comment by janboll** [comment]
Tested locally by resetting my account in dev

File: reconcile/aws_iam_password_reset.py [hunk 2]
```diff
@@ -34,40 +36,66 @@ def account_in_roles(roles: Iterable[Mapping[str, Any]], aws_account: str) -> bo
...
+                logging.error(f"User {user_name} is not in account {account_name}")
+                sys.exit(1)
+
+            accounts_to_reset.append(
+                AwsProfileToReset(
+                    **{
+                        "account": a,
+                        "user_name": user_name,
+                        "state_key": state_key,
+                    }
```
Comment (geoberle) [inline reconcile/aws_iam_password_reset.py:83]: can you pass the arguments directly?

File: reconcile/aws_iam_password_reset.py [hunk 3]
```diff
@@ -34,40 +36,66 @@ def account_in_roles(roles: Iterable[Mapping[str, Any]], aws_account: str) -> bo
...
+                )
+            )
+
+    for a in accounts_to_reset:
+        with AWSApi(1, [a.account], settings=settings) as aws_api:
```
Comment (geoberle) [inline reconcile/aws_iam_password_reset.py:88]: wouldn't this establish a new AWS client for each user password to reset, even if all are in the same AWS account?

Comment (janboll) [reply reconcile/aws_iam_password_reset.py:88]: Do you reset accounts for others? :smiling_imp:

Comment (geoberle) [reply reconcile/aws_iam_password_reset.py:98]: thanks!

**Review by geoberle — APPROVED** [review]
LGTM

**Pull request merged**